RELEASE/scripts/autoscend/auto_post_adv.ash

```diff
@@ -641,20 +641,118 @@ void handlePostAdventure()
+		boolean[location] lowMLZones = $locations[The Smut Orc Logging Camp];
+
+		// Generic Conditions
 		if(get_property("kingLiberated").to_boolean())
 		{
 			doML = false;
 		}
+		if(((get_property("flyeredML").to_int() > 9999) || get_property("auto_hippyInstead").to_boolean() || (get_property("auto_war") == "finished") || (get_property("sidequestArenaCompleted") != "none")) && ((my_level() >= 13)))
+		{
+			doML = false;
+		}
+
+		// Item specific Conditions
 		if((equipped_amount($item[Space Trip Safety Headphones]) > 0) || (equipped_amount($item[Red Badge]) > 0))
 		{
 			doML = false;
+			removeML = true;
 		}
-		if(((get_property("flyeredML").to_int() > 9999) || get_property("auto_hippyInstead").to_boolean() || (get_property("auto_war") == "finished") || (get_property("sidequestArenaCompleted") != "none")) && (my_level() >= 13))
+
+		// Location Specific Conditions
+		if(lowMLZones contains my_location())
 		{
 			doML = false;
-			#auto_change_mcd(0);
+			removeML = true;
+			purgeML = true;
+		}
+		if(highMLZones contains my_location())
+		{
+			doML = true;
+			removeML = false;
+		}
+
+		// Act on ML settings
+		if(doML)
+		{
+			auto_change_mcd(11);
+
+			// Catch when we leave lowMLZone, allow for being "side tracked" buy delay burning
+			if((have_effect($effect[Driving Intimidatingly]) > 0) && (get_property("auto_debuffAsdonDelay") >= 2))
+			{
+				print("No Reason to delay Asdon Usage");
+				uneffect($effect[Driving Intimidatingly]);
+				set_property("auto_debuffAsdonDelay", 0);
+			}
+			else if((have_effect($effect[Driving Intimidatingly]).to_int() == 0)  && (get_property("auto_debuffAsdonDelay") >= 0))
+			{
+				set_property("auto_debuffAsdonDelay", 0);
+			}
+			else
+			{
+				set_property("auto_debuffAsdonDelay", get_property("auto_debuffAsdonDelay").to_int() + 1);
+				print("Delaying debuffing Asdon: " + get_property("auto_debuffAsdonDelay"));
+			}
+
+			if((monster_level_adjustment() + (2 * my_level())) <= 150)
+			{
+				buffMaintain($effect[Ur-Kel\'s Aria of Annoyance], 80, 1, 10);
+			}
+			if((monster_level_adjustment() + 10) <= 150)
+			{
+				buffMaintain($effect[Drescher\'s Annoying Noise], 80, 1, 10);
+			}
+			if((monster_level_adjustment() + 10) <= 150)
+			{
+				buffMaintain($effect[Pride of the Puffin], 80, 1, 10);
+			}
+			if((monster_level_adjustment() + 30) <= 150)
+			{
+				buffMaintain($effect[Ceaseless Snarling], 0, 1, 10);
+			}
 		}
+
+		// If we are in some state where we do not want +ML (Level 13 or Smut Orc) make sure ML is removed
+		if(removeML)
+		{
+			auto_change_mcd(0);
+
+			if(have_effect($effect[Driving Recklessly]) > 0)
+			{
+				uneffect($effect[Driving Recklessly]);
+			}
+			if(0 < have_effect($effect[Ur-Kel\'s Aria of Annoyance]))
+			{
+				uneffect($effect[Ur-Kel\'s Aria of Annoyance]);
+			}
+
+			if((purgeML) && item_amount($item[soft green echo eyedrop antidote]) > 5)
+			{
+				if(0 < have_effect($effect[Drescher\'s Annoying Noise]))
```

Comment:
`uneffect()` already checks if you have the effect, so this block of code can be waaay shorter. :)

Reply:
Awesome thanks!

Fixed.